bot/main.ts

```diff
@@ -0,0 +1,86 @@
+#!/usr/bin/env -S node --no-warnings --loader ts-node/esm
+/**
+ * Wechaty - Conversational RPA SDK for Chatbot Makers.
+ *  - https://github.com/wechaty/wechaty
+ */
+import {
+    Contact,
+    Message,
+    ScanStatus,
+    Wechaty,
+    log,
+} from 'wechaty'
+
+import qrcodeTerminal from 'qrcode-terminal'
+
+// https://stackoverflow.com/a/42817956/1123955
+// https://github.com/motdotla/dotenv/issues/89#issuecomment-587753552
+import 'dotenv/config.js'
+
+function onLogout(user: Contact) {
+    log.info('StarterBot', '%s logout', user)
+}
+
+function onScan(qrcode: string, status: ScanStatus) {
+    if (status === ScanStatus.Waiting || status === ScanStatus.Timeout) {
+        const qrcodeImageUrl = [
+            'https://wechaty.js.org/qrcode/',
+            encodeURIComponent(qrcode),
+        ].join('')
+        log.info('StarterBot', 'onScan: %s(%s) - %s', ScanStatus[status], status, qrcodeImageUrl)
+
+        qrcodeTerminal.generate(qrcode, { small: true })  // show qrcode on console
+
+    } else {
+        log.info('StarterBot', 'onScan: %s(%s)', ScanStatus[status], status)
+    }
+}
+
+function onLogin(user: Contact) {
+    log.info('StarterBot', '%s login', user)
+}
+
+async function onMessage(msg: Message) {
+    log.info('StarterBot', msg.toString())
+
+    if (msg.text() === 'ding') {
```

Comment:
Codacy has a fix for the issue: [Strings must use doublequote.](https://app.codacy.com/gh/philip-zhan/badminton-group/pullRequest?prid=8160404)
```suggestion
    if (msg.text() === "ding") {
```